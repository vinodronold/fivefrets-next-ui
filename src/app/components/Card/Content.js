import React, { PureComponent } from 'react'
import { node } from 'prop-types'
export default class Content extends PureComponent {
  static propTypes = {
    children: node.isRequired
  }
  render() {
    return (
      <div>
        {this.props.children}
        <style jsx>{`
          margin: 0 1rem;
          display: block;
        `}</style>
      </div>
    )
  }
}
