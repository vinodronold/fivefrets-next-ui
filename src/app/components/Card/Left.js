import React, { PureComponent } from 'react'
import { string } from 'prop-types'
import { YTImg } from '../../utils'

export default class Left extends PureComponent {
  static propTypes = {
    id: string.isRequired,
    height: string.isRequired
  }
  render() {
    const { id, height } = this.props
    return (
      <div>
        <img src={YTImg(id)} alt={'YT'} height={height} width={'auto'} />
        <style jsx>{`
          div {
            display: flex;
            margin-right: 1rem;
            align-items: center;
            height: auto;
            width: auto;
          }
        `}</style>
      </div>
    )
  }
}
