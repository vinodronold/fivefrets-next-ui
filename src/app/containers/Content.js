import { connect } from 'react-redux'
import ContentWrapper from '../components/ContentWrapper'
const mapStateToProps = state => ({
  ...state.ui
})

export default connect(mapStateToProps)(ContentWrapper)