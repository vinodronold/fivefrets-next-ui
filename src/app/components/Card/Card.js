import React, { PureComponent } from 'react'
import { string } from 'prop-types'
import { color } from '../../constants/theme'
import Left from './Left'
import Content from './Content'
import { Title, SubHeading1 } from '../Typography'

export default class Card extends PureComponent {
  static propTypes = {
    id: string.isRequired,
    title: string.isRequired,
    subtitle: string,
    height: string
  }

  render() {
    const { id, title, subtitle, height = 'auto' } = this.props
    return (
      <div>
        <Left id={id} height={height} />
        <Content>
          <Title>{title}</Title>
          <SubHeading1>{subtitle}</SubHeading1>
        </Content>
        <style jsx>{`
          div {
            display: flex;
            align-items: center;
            min-height: 3rem;
            margin: 1rem auto;
            background-color: rgba(0, 0, 0, 0.65);
            cursor: pointer;
            transition: all 0.5s;
            width: 80%;
          }
          div:hover {
            color: ${color.secondary(1)};
          }
        `}</style>
      </div>
    )
  }
}
