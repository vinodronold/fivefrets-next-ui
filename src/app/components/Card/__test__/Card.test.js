import React from 'react'
import renderer from 'react-test-renderer'
import Left from '../Left'
import Content from '../Content'
import Card from '../Card'
describe('components/Card.js', () => {
  it('should match the Left spanshot', () => {
    const _left = renderer.create(<Left id={'test_id'} height={'14rem'} />)
    expect(_left).toMatchSnapshot()
  })

  it('should match the Content spanshot', () => {
    const _content = renderer.create(<Content>Content</Content>)
    expect(_content).toMatchSnapshot()
  })

  it('should match the Card spanshot with default height', () => {
    const _card = renderer.create(<Card id={'test_id'} title={'title'} subtitle={'subtitle'} onClick={jest.fn()} />)
    expect(_card).toMatchSnapshot()
  })

  it('should match the Card spanshot with defined height', () => {
    const _card = renderer.create(
      <Card id={'test_id'} title={'title'} subtitle={'subtitle'} height={'20rem'} onClick={jest.fn()} />
    )
    expect(_card).toMatchSnapshot()
  })
})
