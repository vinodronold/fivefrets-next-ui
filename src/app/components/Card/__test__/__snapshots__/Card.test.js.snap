// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Card.js should match the Card spanshot with default height 1`] = `
<div
  className="jsx-2416660906"
>
  <div
    className="jsx-4231049512"
  >
    <img
      alt="YT"
      className="jsx-4231049512"
      height="auto"
      src="https://i.ytimg.com/vi/test_id/default.jpg"
      width="auto"
    />
  </div>
  <div
    className="jsx-250553575"
  >
    <div
      className="jsx-4247411757 jsx-873826105"
      style={Object {}}
    >
      title
    </div>
    <div
      className="jsx-4247411757 jsx-482228210"
      style={Object {}}
    >
      subtitle
    </div>
  </div>
</div>
`;

exports[`components/Card.js should match the Card spanshot with defined height 1`] = `
<div
  className="jsx-2416660906"
>
  <div
    className="jsx-4231049512"
  >
    <img
      alt="YT"
      className="jsx-4231049512"
      height="20rem"
      src="https://i.ytimg.com/vi/test_id/default.jpg"
      width="auto"
    />
  </div>
  <div
    className="jsx-250553575"
  >
    <div
      className="jsx-4247411757 jsx-873826105"
      style={Object {}}
    >
      title
    </div>
    <div
      className="jsx-4247411757 jsx-482228210"
      style={Object {}}
    >
      subtitle
    </div>
  </div>
</div>
`;

exports[`components/Card.js should match the Content spanshot 1`] = `
<div
  className="jsx-250553575"
>
  Content
</div>
`;

exports[`components/Card.js should match the Left spanshot 1`] = `
<div
  className="jsx-4231049512"
>
  <img
    alt="YT"
    className="jsx-4231049512"
    height="14rem"
    src="https://i.ytimg.com/vi/test_id/default.jpg"
    width="auto"
  />
</div>
`;
